### ISSUE TITLE
I would like to know the differences and connections between the amPEPpy tool and https://app.cbbio.online/ampep/method.

### ISSUE THREAD

**Issue by 2842969893** [tlawrence3/amPEPpy#5]
Hello！
 
I am trying to use the amPEPpy tool you provided and the server at https://app.cbbio.online/ampep/method to predict the same batch of amino acid sequence files. I noticed that the results from the two are very different. For over 80,000 amino acid sequences, the website predicts more than 50,000 AMP positives, while amPEPpy predicts only 393 AMP positives.
 
I would sincerely like to ask whether amPEPpy is a new model developed by you, or whether you made corresponding modifications and optimizations based on the model on the website. I mainly want to understand why there is such a big difference between the prediction results of the two.
 
I look forward to your reply and wish you a pleasant day!

**Comment by tlawrence3** [comment]
Hi @2842969893,

When we first set out to make amPEPpy it was to recreate the ampep model but in a more portable format. However, during our initial testing we realized that the original ampep model trained on the original dataset was classifying almost all short peptides as AMPs. This suggested that the model was implicitly learning sequence length and that the training data was biased by having short sequences being mostly AMPs. We confirmed this by showing misclassification rate by sequence length (i.e., very high recall, but very low precision for short sequences) and ratio of AMPs/non-AMPS by sequence length.

We addressed this by balancing the training set by length and AMP/non-AMP sequences making sure that the model was forced to learn peptide features representing AMP activity and not sequence length. This length and class balanced training set led to high accuracy across all sequence lengths and explains the difference between the number of sequences identified as AMPs by the original method and our model. We outline this throughly in the publication here: [https://academic.oup.com/bioinformatics/article/37/14/2058/5948991](https://academic.oup.com/bioinformatics/article/37/14/2058/5948991 )
